nncf/openvino/graph/layer_attributes.py

```diff
@@ -79,51 +81,71 @@ def get_weighted_layer_attributes(
     :param constant_attributes: Constant attributes collected for the given node.
     :return: Weighted layer attributes for the given node.
     """
-    retval = {}
-    for port_id, attrs in constant_attributes.items():
-        if ov_metatype in [
-            OVConvolutionMetatype,
-            OVDepthwiseConvolutionMetatype,
-            OVGroupConvolutionMetatype,
-            OVConvolutionBackpropDataMetatype,
-            OVGroupConvolutionBackpropDataMetatype,
-        ]:
-            node_attrs = ov_node.get_attributes()
-            kwargs = {
-                "weight_requires_grad": False,
-                "stride": tuple(node_attrs["strides"]),
-                "dilations": node_attrs["dilations"],
-                "transpose": ov_metatype in [OVConvolutionBackpropDataMetatype, OVGroupConvolutionBackpropDataMetatype],
-                # TODO: ticket 114378: unify pad attribute
-                "padding_values": tuple(node_attrs["pads_begin"] + node_attrs["pads_end"]),
+    if len(constant_attributes) != 1:
+        return None
+
+    port_id, attrs = constant_attributes.copy().popitem()
+    if ov_metatype in [
+        OVConvolutionMetatype,
+        OVDepthwiseConvolutionMetatype,
+        OVGroupConvolutionMetatype,
+        OVConvolutionBackpropDataMetatype,
+        OVGroupConvolutionBackpropDataMetatype,
+    ]:
+        node_attrs = ov_node.get_attributes()
+        kwargs = {
+            "weight_requires_grad": False,
+            "stride": tuple(node_attrs["strides"]),
+            "dilations": node_attrs["dilations"],
+            "transpose": ov_metatype in [OVConvolutionBackpropDataMetatype, OVGroupConvolutionBackpropDataMetatype],
+            # TODO: ticket 114378: unify pad attribute
+            "padding_values": tuple(node_attrs["pads_begin"] + node_attrs["pads_end"]),
+        }
+
+        weights_layout_map = {
+            OVConvolutionMetatype: [LayoutElem.C_OUT, LayoutElem.C_IN],
+            OVGroupConvolutionMetatype: [LayoutElem.GROUPS, LayoutElem.C_OUT, LayoutElem.C_IN],
+            OVDepthwiseConvolutionMetatype: [LayoutElem.GROUPS, LayoutElem.C_OUT, LayoutElem.C_IN],
+            OVConvolutionBackpropDataMetatype: [LayoutElem.C_IN, LayoutElem.C_OUT],
+            OVGroupConvolutionBackpropDataMetatype: [LayoutElem.GROUPS, LayoutElem.C_IN, LayoutElem.C_OUT],
+        }
```

Comment:
> Also, it's not clear why the layout for `Convolution` contains only `out` and `in` channels. It also has other dimensions. Shouldn't we introduce a full layout for consistency? It might be useful to determine the full layout while building `NNCFGraph` instead of calculating in algorithms like `ChannelAlignment`.

Full layout is being calculated below in this function for both convolutions and matmuls 
Convs:
https://github.com/openvinotoolkit/nncf/blob/0982ce472dfb2f05c70f044710695796f3e835b8/nncf/openvino/graph/layer_attributes.py#L125

MM:
https://github.com/openvinotoolkit/nncf/blob/0982ce472dfb2f05c70f044710695796f3e835b8/nncf/openvino/graph/layer_attributes.py#L131

Reply:
> Can we place this mapping into the metatype attributes?

Done